packages/services/schema/src/lib/compose.ts

```diff
@@ -315,57 +338,78 @@ async function callExternalService(
   });
 
   try {
-    logger?.debug('Calling external composition service (url=%s)', input.url);
+    logger?.debug(
+      'Calling external composition service (url=%s, timeout=%s)',
+      input.url,
+      timeoutMs,
+    );
+
     const response = await got(input.url, {
       method: 'POST',
       headers: input.headers,
       body: input.body,
       responseType: 'text',
       retry: {
-        limit: 5,
+        limit: 3,
         methods: ['POST', ...(got.defaults.options.retry.methods ?? [])],
         statusCodes: [404].concat(got.defaults.options.retry.statusCodes ?? []),
         backoffLimit: 500,
       },
       timeout: {
         request: timeoutMs,
         // connecting should be quick
-        lookup: 10_000,
-        connect: 10_000,
-        secureConnect: 10_000,
+        lookup: 5_000,
+        connect: 5_000,
+        secureConnect: 5_000,
```

Comment:
We have
`SCHEMA_COMPOSITION_TIMEOUT_MS` - Timeout for the whole composition process (main process kills worker)
`SCHEMA_EXTERNAL_COMPOSITION_TIMEOUT_MS` - Timeout for a external composition call

`SCHEMA_COMPOSITION_TIMEOUT_MS` must be higher than `SCHEMA_EXTERNAL_COMPOSITION_TIMEOUT_MS * EXTERNAL_COMPOSITION_RETRY_COUNT`.

```
EXTERNAL_COMPOSITION_RETRY_COUNT -> 9_000
EXTERNAL_COMPOSITION_RETRY_COUNT -> 3
TOTAL = 27_000
```

That means `SCHEMA_COMPOSITION_TIMEOUT_MS` set to `30_000` is okay. That way we get the actual HTTP/Socket error we can show to the end user (for their own external composition service). Instead of a generic schema composition timed out or unexpected error.